### DISCUSSION TITLE
Collaboration Thread

### DISCUSSION THREAD

**Discussion by Liftoff-Studios** [Liftoff-KA/Ultimate-KA-Extension#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Liftoff-Studios** [comment]
Spencer pls report here :)

**Reply by SpencerKuan** [reply]
Hello, I'm here :)

**Comment by Liftoff-Studios** [comment]
Hello :)

**Comment by Liftoff-Studios** [comment]
So let us first decide the features

**Comment by Liftoff-Studios** [comment]
Some features we could add

- Dark Mode
- Unvote button
- Support for other languages

**Reply by SpencerKuan** [reply]
Multi-language seems like it would be difficult, but it would be useful. Would the code be translated into PJS or HTML using a library? 

Two other possible features:
* Automatic notification updates, if that's possible
* Showing program flags

**Reply by Liftoff-Studios** [reply]
Use languages would be difficult. I will take a lot of time to figure out how to do that. One way might be to change the settings of the ace editor

**Comment by Liftoff-Studios** [comment]
Wow these are good. So I think we can work on these first

**Comment by Liftoff-Studios** [comment]
I'll work on multi language, dark mode and unvote button.

**Comment by Liftoff-Studios** [comment]
Ok. I made the unvote button code. It is in another file though and needs some finishing touches. I'll do those and merge the  code 😄

**Comment by Liftoff-Studios** [comment]
Is autonotif.js working. if it is pls merge the code 😃

**Comment by SpencerKuan** [comment]
New autonotif code looks good :)

**Reply by Liftoff-Studios** [reply]
Thanks :D

**Reply by Liftoff-Studios** [reply]
Should  I delete the old code ??

**Reply by SpencerKuan** [reply]
Sure

**Comment by Liftoff-Studios** [comment]
Are you working on anything ?
Could you work on the dark mode ?

**Comment by Liftoff-Studios** [comment]
Could you work on showing the flags ?

**Comment by SpencerKuan** [comment]
Sorry, Liftoff, I’ve been pretty busy with homework recently. I’ll work on the flags when I have time. 

I tried to rewrite the code in the branch test to see if I could get some of the code to run as an extension, but the KA API is giving me errors. Am I misunderstanding how your code is supposed to function? Im packing the code as an extension on chrome.

**Reply by Liftoff-Studios** [reply]
No problem. Life comes first :D
Since we haven't mentioned the scripts in manifest. Json it n may not be running

For a quick test just paste the code in the console

**Comment by SpencerKuan** [comment]
Do you want to start working on this again?

VEXCESS created some documentation for the new API

https://www.khanacademy.org/computer-programming/new-ka-api/5236998489030656

**Reply by Liftoff-Studios** [reply]
Yes

**Reply by Liftoff-Studios** [reply]
We could continue this 👍

**Comment by Liftoff-Studios** [comment]
Also do you have a replit account ?

**Reply by SpencerKuan** [reply]
I do, albeit an inactive one. 
https://replit.com/@SpencerKuan

**Comment by Liftoff-Studios** [comment]
Ah great !

**Comment by Liftoff-Studios** [comment]
The problem is the api for voting hasn't been replaced yet :(

**Reply by SpencerKuan** [reply]
It hasn’t? Interesting 

We could try to actually load the entire KA page, and then traverse the DOM to grab the votes, but that would be really slow.

**Reply by Liftoff-Studios** [reply]
No we would have to wait for the vote api to change

**Comment by Liftoff-Studios** [comment]
This is my account: 
https://replit.com/@Liftoff-KA
